src/index.js

```diff
@@ -0,0 +1,311 @@
+    const connectionTest = await apiClient.testConnection();
+    if (!connectionTest.success) {
+      logMessage(logFile, `API connectivity test failed: ${connectionTest.error}`, 'ERROR');
+      throw new Error(`API connectivity test failed: ${connectionTest.error}`);
+    }
+    core.info('✅ API connectivity confirmed');
+    logMessage(logFile, 'API connectivity confirmed');
+    
+    // Launch the workflow
+    core.info('🎯 Launching workflow...');
+    logMessage(logFile, 'Launching workflow...');
+    const launchResult = await apiClient.launchWorkflow(inputs);
+    
+    if (!launchResult.success) {
+      // Provide detailed error information
+      let errorMessage = `Workflow launch failed: ${launchResult.error}`;
+      
+      if (launchResult.statusCode === 401) {
+        errorMessage += '\n\n💡 This usually indicates an invalid or expired access token.';
+        errorMessage += '\n   Please check that your TOWER_ACCESS_TOKEN secret is valid.';
+      } else if (launchResult.statusCode === 403) {
+        errorMessage += '\n\n💡 This usually indicates insufficient permissions.';
+        errorMessage += '\n   Please check workspace permissions and compute environment access.';
+      } else if (launchResult.statusCode === 404) {
+        errorMessage += '\n\n💡 This usually indicates the pipeline or workspace was not found.';
+        errorMessage += '\n   Please check the pipeline URL and workspace ID.';
+      }
+      
+      if (launchResult.details && inputs.debug) {
+        errorMessage += `\n\n🐛 Debug details: ${launchResult.details}`;
+      }
+      
+      throw new Error(errorMessage);
+    }
+    
+    const workflowData = launchResult.data;
+    core.info(`✅ Workflow launched successfully!`);
+    core.info(`📊 Workflow ID: ${workflowData.workflowId}`);
+    logMessage(logFile, `Workflow launched successfully! ID: ${workflowData.workflowId}`);
+    
+    // Build workflow URL (this might need adjustment based on actual API responses)
+    let workflowUrl = workflowData.workflowUrl;
+    if (!workflowUrl) {
+      // Construct URL if not provided by API
+      const baseUrl = inputs.apiEndpoint.replace('/api.', '/').replace('/api', '');
+      if (inputs.workspaceId) {
+        workflowUrl = `${baseUrl}/orgs/-/workspaces/${inputs.workspaceId}/watch/${workflowData.workflowId}`;
+      } else {
+        workflowUrl = `${baseUrl}/workflow/${workflowData.workflowId}`;
+      }
+    }
+    
+    // Set outputs
+    const outputs = {
+      workflowId: workflowData.workflowId,
+      workflowUrl: workflowUrl,
+      workspaceId: workflowData.workspaceId || inputs.workspaceId || '',
+      workspaceRef: workflowData.workspaceRef || '[personal]',
+      json: JSON.stringify({
+        workflowId: workflowData.workflowId,
+        workflowUrl: workflowUrl,
+        workspaceId: workflowData.workspaceId || inputs.workspaceId || '',
+        workspaceRef: workflowData.workspaceRef || '[personal]'
+      })
+    };
```

Comment:
Not sure if this is valid JS, but seems a bit duplicative..
```suggestion
    };
    outputs.json = JSON.stringify(outputs);
```